code/android-core-library/src/main/java/com/adobe/marketing/mobile/util/TimeUtils.kt

```diff
@@ -36,45 +39,69 @@ object TimeUtils {
+     * @return date [String] formatted as ISO 8601 with RFC 822 time zone, second precision
      */
     @JvmStatic
-    fun getIso8601Date(): String? {
-        return getIso8601Date(Date(), ISO8601_DATE_FORMATTER_TIMEZONE_RFC822)
+    @JvmOverloads
+    fun getIso8601DateTimeZoneRFC822(date: Date? = Date()): String? {
+        return getFormattedDate(date, DatePattern.ISO8601_TIMEZONE_RFC822_PRECISION_SECOND)
     }
 
+
     /**
-     * Gets the the ISO 8601 formatted date `String` for the current date.
-     * TimeZone format used is ISO8601 using XXX formatting letters. ex: 9th July 2020 PDT will be formatted as 2020-07-09T15:09:18-07:00.
+     * Gets the ISO 8601 formatted with UTC(Z) time zone, millisecond precision date `String` for the
+     * provided date using the UTC +0 time zone.
+     * Date pattern used is [DatePattern.ISO8601_TIMEZONE_ISO8601_UTCZ_PRECISION_MILLISECOND]
+     * which has timezone ISO 8601 'Z' char terminator, which gives a timezone of 'Z'.
+     * Ex: Wed Nov 30 11:53:09.497 GMT-07:00 2022 -> 2022-11-30T18:53:09.497Z (notice the hour shift because the date must be evaluated from UTC +0)
+     *
+     * Note that ISO 8601 requires date strings terminating with the char 'Z' to be in the time zone UTC +0 (no offset).
+     *
      * AMSDK-10273 :: ExEdge requires time zone offset formatted in form [+-]HH:MM.
      *
-     * @return Iso8601 formatted date [String]
+     * @param date the [Date] to apply the formatting to; defaults to the current [Date]
+     * @return date [String] formatted as ISO 8601 with UTC(Z) time zone, millisecond precision
      */
     @JvmStatic
-    fun getIso8601DateTimeZoneISO8601(): String? {
-        return getIso8601Date(Date(), ISO8601_DATE_FORMATTER_TIMEZONE_UTC, TimeZone.getTimeZone("GMT"))
+    @JvmOverloads
+    fun getIso8601DateTimeZoneUTC(date: Date? = Date()): String? {
+        return getFormattedDate(date, DatePattern.ISO8601_TIMEZONE_ISO8601_UTCZ_PRECISION_MILLISECOND, TimeZone.getTimeZone("Etc/UTC"))
     }
 
     /**
-     * Gets the the ISO 8601 formatted date `String` for the passed in date - "yyyy-MM-dd'T'HH:mm:ssZZZ"
+     * Gets the formatted date `String` for the passed in date; if no date passed in, uses current `Date()`
      *
-     * @param date the [Date] to generate the [String] for
-     * @return ISO 8601 formatted date [String]
+     * @param date the [Date] to apply the formatting to; defaults to the current [Date]
+     * @param datePattern the [DatePattern] to use to format the date [String]
+     * @param timeZone the [TimeZone] to evaluate the formatted date from; defaults to device time zone if not specified
+     * @return the formatted date [String]
      */
     @JvmStatic
     @JvmOverloads
-    fun getIso8601Date(date: Date?, format: String?, timeZone: TimeZone? = null): String? {
+    fun getFormattedDate(date: Date?, datePattern: DatePattern, timeZone: TimeZone? = null): String? {
```

Comment:
Can you make this private for now? We can expose this later if any extension needs access to it.

Reply:
Remove these annotations for non public methods.